lib/plugins/aws/customResources/resources/eventBridge/handler.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+const { addPermission, removePermission } = require('./lib/permissions');
+const {
+  createEventBus,
+  deleteEventBus,
+  updateRuleConfiguration,
+  updateTargetConfiguration,
+  removeRuleConfiguration,
+  removeTargetConfiguration,
+} = require('./lib/eventBridge');
+const { getEnvironment, getLambdaArn, handlerWrapper } = require('../utils');
+
+function handler(event, context) {
+  if (event.RequestType === 'Create') {
+    return create(event, context);
+  } else if (event.RequestType === 'Update') {
+    return update(event, context);
+  } else if (event.RequestType === 'Delete') {
+    return remove(event, context);
+  }
+  throw new Error(`Unhandled RequestType ${event.RequestType}`);
+}
+
+function create(event, context) {
+  const { FunctionName, EventBridgeConfig } = event.ResourceProperties;
+  const { Region, AccountId } = getEnvironment(context);
+
+  const lambdaArn = getLambdaArn(Region, AccountId, FunctionName);
+
+  return addPermission({
+    functionName: FunctionName,
+    region: Region,
+    accountId: AccountId,
+    eventBus: EventBridgeConfig.EventBus,
+    ruleName: EventBridgeConfig.RuleName,
+  })
+    .then(() =>
```

Comment:
I know we can't use `async`/`await` in sls code itself, but can't a lambda use it since we control the runtime it's deploying with?

Reply:
Yes, I did the same in the intial PR for existing S3 bucket. @medikoo pointed out that there's some potential rejected promises might not be propagated correctly and he advised that I should use plain Promises rather than `async` / `await`.

Reply:
My original point was that we should use _callback_ resolution instead of _thenable_ lambda resolution, as in latter case AWS env doesn't wait for orphaned async flows to end, and they may leak to further invocations.

Technically we can use async/await as long as top level lambda handler function is not an async function, and we propagate async functions result to AWS callback.